### PR TITLE
Parquet support

### PR DESCRIPTION
**Pull request by shz117**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -526,6 +526,7 @@ project(":samza-hdfs_$scalaVersion") {
       exclude module: 'servlet-api'
       exclude module: 'zookeeper'
     }
+    compile('org.apache.parquet:parquet-hadoop:1.9.0')
```
Comment (atschabu) [inline build.gradle]: looks like version numbers are all defined in ./gradle/dependency-versions.gradle

File: samza-hdfs/src/main/java/org/apache/samza/system/hdfs/reader/ParquetFileHdfsReader.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        offset = 0;
+        // TODO: seek updates internal state lastGroup singleFileOffset times, could be avoid...
+        for (int i = 0; i < bootstrapOffset; i++) {
+            if (!hasNext()) break;
+        }
+    }
+
+    @Override
+    public IncomingMessageEnvelope readNext() {
+        return new IncomingMessageEnvelope(systemStreamPartition, Integer.toString(offset++), null, lastGroup);
```
Comment (atschabu) [inline samza-hdfs/src/main/java/org/apache/samza/system/hdfs/reader/ParquetFileHdfsReader.java]: check hasNext() first. Throw `NoSuchElementException()` if false.

File: samza-hdfs/src/test/java/org/apache/samza/system/hdfs/reader/TestParquetFileHdfsReader.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.apache.samza.system.SystemStreamPartition;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by hongzhengshi on 3/14/17.
+ */
+public class TestParquetFileHdfsReader {
```
Comment (atschabu) [inline samza-hdfs/src/test/java/org/apache/samza/system/hdfs/reader/TestParquetFileHdfsReader.java:37]: add test case, checking behaviour when calling next() when no more elements available.

**Review by atschabu — CHANGES_REQUESTED** [review]

File: samza-hdfs/src/test/java/org/apache/samza/system/hdfs/reader/TestParquetFileHdfsReader.java (hunk 3 above)
Comment (shz117) [reply samza-hdfs/src/test/java/org/apache/samza/system/hdfs/reader/TestParquetFileHdfsReader.java:37]: added testNextThrowsExceptionWhenNoMoreElement below

**Review by atschabu — APPROVED** [review]

**Pull request merged**